classes/galley/DAO.inc.php

```diff
@@ -0,0 +1,308 @@
+ *
+ * @class galley
+ *
+ * @brief Read and write galleys to the database.
+ */
+
+namespace PKP\galley;
+
+use APP\facades\Repo;
+use APP\publication\Publication;
+use Illuminate\Support\Collection;
+use Illuminate\Support\Facades\DB;
+use Illuminate\Support\LazyCollection;
+use PKP\core\EntityDAO;
+use PKP\db\DAOResultFactory;
+use PKP\identity\Identity;
+use PKP\services\PKPSchemaService;
+use PKP\submission\PKPSubmission;
+use PKP\submission\Representation;
+use PKP\submission\RepresentationDAOInterface;
+
+class DAO extends EntityDAO implements RepresentationDAOInterface
+{
+    /** @copydoc EntityDAO::$schema */
+    public $schema = PKPSchemaService::SCHEMA_GALLEY;
+
+    /** @copydoc EntityDAO::$table */
+    public $table = 'publication_galleys';
+
+    /** @copydoc EntityDAO::$settingsTable */
+    public $settingsTable = 'publication_galley_settings';
+
+    /** @copydoc EntityDAO::$primarykeyColumn */
+    public $primaryKeyColumn = 'galley_id';
+
+    /** @copydoc EntityDAO::$primaryTableColumns */
+    public $primaryTableColumns = [
+        'submissionFileId' => 'submission_file_id',
+        'id' => 'galley_id',
+        'isApproved' => 'is_approved',
+        'locale' => 'locale',
+        'label' => 'label',
+        'publicationId' => 'publication_id',
+        'seq' => 'seq',
+        'urlPath' => 'url_path',
+        'urlRemote' => 'remote_url',
+        'doiId' => 'doi_id',
+    ];
+
+    public function newDataObject(): Galley
+    {
+        return app(Galley::class);
+    }
+
+    public function get(int $id): ?Galley
+    {
+        return parent::get($id);
+    }
+
+    public function getByUrlPath(string $urlPath, Publication $publication): ?Galley
+    {
+        $row = DB::table($this->table)
+            ->where('publication_id', $publication->getId())
+            ->where('url_path', $urlPath)
+            ->first();
+        return $row ? $this->fromRow($row) : null;
+    }
+
+    /**
+     * Get the number of galleys matching the configured query
+     */
+    public function getCount(Collector $query): int
+    {
+        return $query
+            ->getQueryBuilder()
+            ->select('g.' . $this->primaryKeyColumn)
+            ->get()
+            ->count();
+    }
+
+    /**
+     * Get a list of ids matching the configured query
+     */
+    public function getIds(Collector $query): Collection
+    {
+        return $query
+            ->getQueryBuilder()
+            ->select('g.' . $this->primaryKeyColumn)
+            ->pluck('g.' . $this->primaryKeyColumn);
+    }
+
+    /**
+     * Get a collection of galleys matching the configured query
+     */
+    public function getMany(Collector $query): LazyCollection
+    {
+        $rows = $query
+            ->getQueryBuilder()
+            ->select(['g.*'])
+            ->get();
+
+        return LazyCollection::make(function () use ($rows) {
+            foreach ($rows as $row) {
+                yield $this->fromRow($row);
```

Comment:
Sorry, I know we came down on opposite sides of this, but I still think we should be yielding by IDs :)

Reply:
I'm on board, now! :) I've stumbled over it a few times but always been able to solve it quickly.